### DISCUSSION TITLE
npx directus restart/reboot = unknown command

### DISCUSSION THREAD

**Discussion by jjaimealeman** [directus/directus#12637] [Archive]
### Describe the Bug

After recently making an edit to my server `.env` file, I could not restart Directus.
As this is a DigitalOcean droplet, I had to reboot the server to reflect the recent change in .env.

Is there a better solution?

### To Reproduce

SSH into server.

Run command `npx directus restart` or `npx directus reboot`

Available commands are:
![screenshot](https://i.imgur.com/zoSRJyP.png)

### Errors Shown

```
# npx directus restart
error: unknown command 'restart'

# npx directus reboot
error: unknown command 'reboot'
```

### What version of Directus are you using?

9.7.1

### What version of Node.js are you using?

14.19.1

### What database are you using?

unknown

### What browser are you using?

Firefox Dev 98.0.2

### How are you deploying Directus?

Digital Ocean Droplet

**Comment by rijkvanzanten** [comment]
`directus reboot` / `directus restart` aren't Directus commands that exist, so the error you're seeing there make sense. How are you running the Directus process initially? If this is based on the one-click, I believe Directus is ran by the `node` user through `pm2`. Try `pm2 list` or `su node && pm2 list`. If you see the Directus process in either of those, you should be able to restart the process with `pm2 restart <pid>` 👍🏻

**Reply by jjaimealeman** [reply]
Yes, this is the DigitalOcean One-Click Installer on Ubuntu 20.04.4 LTS.

Here is the result of running `pm2 list` `su node && pm2 list`
![screenshot](https://i.imgur.com/kASBZ3r.png)

`su node` resulted in error `su: user node does not exist` so I ran `awk -F: '{ print $1}' /etc/passwd` to list users:
![screenshot](https://i.imgur.com/14A099T.png)

**Reply by rijkvanzanten** [reply]
Ah try it as the directus user. Change into it with `su directus`, and then run `pm2 list` 👍🏻

**Reply by jjaimealeman** [reply]
![screenshot](https://i.imgur.com/cV66ojD.png)

😒

**Comment by sawyerclick** [comment]
I'm not sure if it's the most elegant way but this is working for me to restart the app, including with updated environment variables: 

```bash
npx directus pm2 restart app --update-env
```

I believe the only way to access the directus user is through the `npx directus pm2` command.

**Reply by jjaimealeman** [reply]
sorry for the late reply.

![screenshot](https://i.imgur.com/wGXQawb.png)

I haven't messed around with this in a while.

This brings to mind another question (unrelated): what are the more common methods of installation as I feel the DO One-Click is not common.

**Reply by sawyerclick** [reply]
**If you're using the DO one-click, the process is probably named `directus`**

Sorry, I should have specified that `app` could be anything and should be re-fitted to reflect the name of your process. Find that by running `npx directus pm2 ls`. This is what I run to restart my Directus DO one-click droplet and is probably what will work for you.

```bash
npx directus pm2 restart directus --update-env
```


As for what people are commonly using, I'm not sure. I've defaulted to the one-click but have found it to not be very common. I use it just locally in a few other instances where I'm SSG-ing a site. I've been eager to check out using [Directus with Railway](https://github.com/directus/directus/discussions/9626) but haven't gotten around to it yet

**Reply by jjaimealeman** [reply]
![screenshot](https://i.imgur.com/nQ2w8jl.png)

👍

Heck yeah buddy, that did the trick!
Thank you!

Closing this issue.
Thank you all for your time and effort!
😁

**Reply by sawyerclick** [reply]
Awesome! 🚀  Make sure to tag this one as the right answer so future Directus-ers can find it 🙏

**Comment by ashiva** [comment]
Using the latest DO 1 click, after updating the .env file and running `npx directus pm2 restart directus --update-env` in `\var\directus` I get `error: unknown command 'pm2'` - any ideas?

**Reply by sawyerclick** [reply]
Does running it as `pm2 restart directus --update-env` do anything? Does typing just `pm2` display anything?

Lately I've found I didn't need the `npx directus` directives. Not sure what changed there.
